services/publicapi/service.go

```diff
@@ -49,12 +52,27 @@ func NewPublicApi(
 
 func (s *service) HandleTransactionResults(input *handlers.HandleTransactionResultsInput) (*handlers.HandleTransactionResultsOutput, error) {
 	for _, txReceipt := range input.TransactionReceipts {
-		s.txWaiter.reportCompleted(txReceipt, input.BlockHeight, input.Timestamp)
+		s.waiter.complete(txReceipt.Txhash().KeyForMap(),
+			&waiterObject{&txResponse{
+				transactionStatus:  protocol.TRANSACTION_STATUS_COMMITTED,
+				transactionReceipt: txReceipt,
+				blockHeight:        input.BlockHeight,
+				blockTimestamp:     input.Timestamp,
+			}})
 	}
 	return &handlers.HandleTransactionResultsOutput{}, nil
 }
 
 func (s *service) HandleTransactionError(input *handlers.HandleTransactionErrorInput) (*handlers.HandleTransactionErrorOutput, error) {
+	//s.waiter.complete(input.Txhash.KeyForMap(),
```

Comment:
Why don't you implement this?

Reply:
shai's code is not in yet - there is no tx hash so i cannot actually use my code.